### PR TITLE
0.5 holder contract

### PR DESCRIPTION
**Pull request by i-stam**
- Claim function for non-redeemable tokens (list of addresses)

- get redeemableTokens  is implemented in the TokenWhitelist contract instead (save gas + easier to change implementation)

### CODE REVIEW

File: contracts/holder.sol [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+            require(!_isTokenBurnable(_nonBurnableAddresses[i]), "burnables cannot be claimed");
+            uint claimBalance = balance(_nonBurnableAddresses[i]);
+            if (claimBalance > 0) {
+                _claim(_to, _nonBurnableAddresses[i], claimBalance);
+            }
+        }
+        return true;
+    }
+
+    function burner() external view returns (address) {
```
Comment (draganm) [inline contracts/holder.sol]: why not declaring `_burner` field public?

File: contracts/internals/tokenWhitelist.sol [hunk 2]
```diff
@@ -181,12 +193,33 @@ contract TokenWhitelist is ENSResolvable, Controllable, Ownable, Claimable {
...
-    /// @return address[] of our whitelisted tokens
+    /// @notice This returns an array of all whitelisted token addresses
+    /// @return address[] of whitelisted tokens
     function tokenAddressArray() external view returns (address[] memory) {
         return _tokenAddressArray;
     }
 
+    /// @notice This returns an array of all burnable token addresses
+    /// @return address[] of burnable tokens
+    function burnableTokens() external view returns (address[] memory) {
```
Comment (draganm) [inline contracts/internals/tokenWhitelist.sol]: we could consider storing an array of burnable tokens instead, but this works ...

File: test/internals/token_whitelist/remove_tokens_test.go [hunk 3]
```diff
@@ -95,6 +90,44 @@ var _ = Describe("removeTokens", func() {
...
+                })
+
+                Context("When removing another supported token", func() {
+
+    				BeforeEach(func() {
+    					tx, err := TokenWhitelist.RemoveTokens(Controller.TransactOpts(), []common.Address{common.HexToAddress("0x3")})
+    					Expect(err).ToNot(HaveOccurred())
+    					Backend.Commit()
+                        Expect(isSuccessful(tx)).To(BeTrue())
+    				})
```
Comment (draganm) [inline test/internals/token_whitelist/remove_tokens_test.go]: indentation is weird

File: test/internals/token_whitelist/remove_tokens_test.go [hunk 4]
```diff
@@ -95,6 +90,44 @@ var _ = Describe("removeTokens", func() {
...
+                    })
+                })
+
+                Context("When removing an unsupported token", func() {
+
+    				BeforeEach(func() {
+    					tx, err := TokenWhitelist.RemoveTokens(Controller.TransactOpts(ethertest.WithGasLimit(100000)), []common.Address{common.HexToAddress("0x5")})
+    					Expect(err).ToNot(HaveOccurred())
+    					Backend.Commit()
+                        Expect(isSuccessful(tx)).To(BeFalse())
```
Comment (draganm) [inline test/internals/token_whitelist/remove_tokens_test.go]: same here - something went wrong with indentation ...

File: test/holder/burn_decimal_test.go [hunk 5]
```diff
@@ -0,0 +1,269 @@
...
+					Expect(evt.From).To(Equal(BankAccount.Address()))
+					Expect(evt.To).To(Equal(RandomAccount.Address()))
+					Expect(evt.Value.String()).To(Equal("187"))
+				})
+
+				When("The holder contract owns two types of ERC20 tokens and 1 ETH", func() {
+
+                    //add the tokens to the list
+					BeforeEach(func() {
+                        tokens := []common.Address{common.HexToAddress("0x0"), ERC20Contract1Address, ERC20Contract2Address}
```
Comment (draganm) [inline test/holder/burn_decimal_test.go]: indentation ...

File: contracts/wallet.sol [hunk 6]
```diff
@@ -722,6 +722,7 @@ contract Wallet is ENSResolvable, Vault, GasTopUpLimit, LoadLimit {
         }
     }
 
+
```
Comment (nostdm) [inline contracts/wallet.sol]: Great change

File: contracts/internals/tokenWhitelist.sol (hunk 2 above)
Comment (i-stam) [reply contracts/internals/tokenWhitelist.sol]: yes, this is true and we need to do it for scaling reasons, that's why we are just exposing an interface now to the holder contract, so we can replace the tokenWhitelist anytime without needing to change the Holder contract

File: test/holder/burn_decimal_test.go (hunk 5 above)
Comment (i-stam) [reply test/holder/burn_decimal_test.go]: funny thing, it must be a github thing, no indentation issue in my editor...

**Pull request merged**